### PR TITLE
feat: add new chains to xdefi

### PR DESCRIPTION
**Pull request by RyukTheCoder**
# Summary

Attempted to add new chains to XDEFI wallet. The status of each chain is indicated below.

- Dash: Not supported!
- Doge: Added get accounts and signer. Tested by swapping from DOGE Doge to RUNE Thorchain.
- Maya: Added get accounts and signer. Tested by swapping from CACAO Maya to RUNE Thorchain.
- Tron: Could not find any injected provider.
- Cosmos: Added get accounts and signer. Tested by swapping from OSMO Osmosis to ATOM Cosmos.

Fixes RF-1286

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by samobasquiat — CHANGES_REQUESTED** [review]

**Comment by samobasquiat** [comment]
In the task, I saw the dash chain as well, but it seems xdefi doesn't support it. Is it true?

**Comment by samobasquiat** [comment]
Some of my tests:
Doge => Rune https://runescan.io/tx/2e5fc11385a4b76670ce576784b2fb7b796ae7a22d9aaeda16387f831fcba259
Rune =>  Doge https://runescan.io/tx/EB64CDF8127BA1B501693AF0BE14314806CAB0764E08E9C6AD463BDF2AA620B6
Atom => Osmo https://www.mintscan.io/osmosis/tx/da3dca0610df9ef08bd46b5749503575dae3e8fafc6a1fd41e74c444bcf86a0c

File: wallets/provider-xdefi/src/index.ts [hunk 1]
```diff
@@ -50,7 +53,36 @@ export const connect: Connect = async ({ instance, meta }) => {
     meta,
   });
 
-  return [evmResult, ...nonEvmResults, solanaAccounts as ProviderConnectResult];
+  const cosmosAccounts = [];
+  if (cosmosInstance) {
+    const supportedCosmosChainIds =
+      await window.xfi?.cosmos?.getSupportedChains?.();
```
[the remark this answers is not in the document]
Comment (RyukTheCoder) [reply wallets/provider-xdefi/src/index.ts:59]: `cosmosInstance` is equal to `window.xfi.keplr` so we can not do that. `window.xfi?.cosmos` can be used to get cosmos supported chains.

[the remark this answers is not in the document]
Comment (RyukTheCoder) [reply wallets/provider-xdefi/src/index.ts]: removed it and used hard coded cosmos chains.

File: wallets/provider-xdefi/src/index.ts [hunk 2]
```diff
@@ -100,22 +132,32 @@ export const canEagerConnect: CanEagerConnect = async ({ instance, meta }) => {
...
-  supportedChains: allBlockChains.filter((blockchainMeta) =>
-    [
-      ...SUPPORTED_ETH_CHAINS,
-      ...XDEFI_WALLET_SUPPORTED_NATIVE_CHAINS,
-      Networks.SOLANA,
-    ].includes(blockchainMeta.name as Networks)
-  ),
-});
+) => {
+  const cosmos = cosmosBlockchains(allBlockChains).filter(
```
[the remark this answers is not in the document]
Comment (RyukTheCoder) [reply wallets/provider-xdefi/src/index.ts]: fixed. used hard coded cosmos chains.

**Review by samobasquiat — APPROVED** [review]

File: wallets/provider-xdefi/src/index.ts [hunk 3]
```diff
@@ -50,7 +53,33 @@ export const connect: Connect = async ({ instance, meta }) => {
     meta,
   });
 
-  return [evmResult, ...nonEvmResults, solanaAccounts as ProviderConnectResult];
+  const cosmosAccounts = [];
```
Comment (Ikari-Shinji-re) [inline wallets/provider-xdefi/src/index.ts:56]: You can specify a type for this array:
```ts
const cosmosAccounts: ProviderConnectResult[] = [];
```

File: wallets/provider-xdefi/src/index.ts [hunk 4]
```diff
@@ -50,7 +53,33 @@ export const connect: Connect = async ({ instance, meta }) => {
...
+  if (cosmosInstance) {
+    const cosmosBlockchainMeta = meta
+      .filter(isCosmosBlockchain)
+      .filter((blockchainMeta: BlockchainMeta) =>
+        SUPPORTED_COSMOS_CHAINS.includes(blockchainMeta.name as Networks)
+      );
```
Comment (Ikari-Shinji-re) [inline wallets/provider-xdefi/src/index.ts:62]: Could you combine these filters and eliminate unnecessary iteration?

File: wallets/provider-xdefi/src/index.ts [hunk 5]
```diff
@@ -100,22 +129,34 @@ export const canEagerConnect: CanEagerConnect = async ({ instance, meta }) => {
...
-      ...XDEFI_WALLET_SUPPORTED_NATIVE_CHAINS,
-      Networks.SOLANA,
-    ].includes(blockchainMeta.name as Networks)
-  ),
-});
+) => {
+  const supportedCosmosChains = cosmosBlockchains(allBlockChains)
+    .filter((blockchainMeta) => !!blockchainMeta.info)
+    .filter((blockchainMeta: BlockchainMeta) =>
+      SUPPORTED_COSMOS_CHAINS.includes(blockchainMeta.name as Networks)
```
Comment (Ikari-Shinji-re) [inline wallets/provider-xdefi/src/index.ts:136]: Could you combine these filters and eliminate unnecessary iteration?

File: wallets/provider-xdefi/src/index.ts (hunk 3 above)
Comment (RyukTheCoder) [reply wallets/provider-xdefi/src/index.ts:56]: Fixed

File: wallets/provider-xdefi/src/index.ts (hunk 4 above)
Comment (RyukTheCoder) [reply wallets/provider-xdefi/src/index.ts:62]: Thanks Fixed

File: wallets/provider-xdefi/src/index.ts (hunk 5 above)
Comment (RyukTheCoder) [reply wallets/provider-xdefi/src/index.ts:136]: Thanks Fixed

**Review by Ikari-Shinji-re — APPROVED** [review]

**Pull request merged**